internal/provider/resource_kubernetes_cloud.go

```diff
@@ -0,0 +1,233 @@
+			"parentcloudname": schema.StringAttribute{
+				Description: "The parent cloud name in case adding k8s cluster from existed cloud. Changing this value will cause the cloud to be destroyed and recreated by terraform.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"parentcloudregion": schema.StringAttribute{
+				Description: "The parent cloud region name in case adding k8s cluster from existed cloud. Changing this value will cause the cloud to be destroyed and recreated by terraform.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+// Create adds a new kubernetes cloud to controllers used now by Terraform provider.
+func (o *kubernetesCloudResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Prevent panic if the provider has not been configured.
+	if o.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "ssh_key", "create")
+		return
+	}
+
+	var plan kubernetesCloudResourceModel
+
+	// Read Terraform configuration from the request into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the kubernetes cloud.
+	err := o.client.Clouds.CreateKubernetesCloud(
+		&juju.CreateKubernetesCloudInput{
+			Name:             plan.CloudName.ValueString(),
+			KubernetesConfig: plan.KubernetesConfig.ValueString(),
+		},
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create kubernetes cloud, got error %s", err))
+		return
+	}
+
+	o.trace(fmt.Sprintf("Created kubernetes cloud %s", plan.CloudName.ValueString()))
+
+	plan.ID = types.StringValue(newKubernetesCloudID(plan.CloudName.ValueString()))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func newKubernetesCloudID(name string) string {
+	return fmt.Sprintf("kubernetes-cloud:%s", name)
+}
+
+// Read reads the current state of the kubernetes cloud.
+func (o *kubernetesCloudResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Prevent panic if the provider has not been configured.
+	if o.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "ssh_key", "read")
+		return
+	}
+
+	var plan kubernetesCloudResourceModel
+
+	// Read Terraform configuration from the request into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read the kubernetes cloud.
+	cloud, err := o.client.Clouds.ReadKubernetesCloud(
+		&juju.ReadKubernetesCloudInput{
+			Name:             plan.CloudName.ValueString(),
+			KubernetesConfig: plan.KubernetesConfig.ValueString(),
+		},
+	)
```

Comment:
issue: Import will not work if Read is implemented this way. You have to get the name from the ID.